frontend/src/app/ta-application/ta-application.service.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * The TA Application Service abstracts backend calls from the
+ * Admin organization List Component.
+ *
+ * @author Ben Goulet
+ * @copyright 2024
+ * @license MIT
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { BehaviorSubject, Observable, switchMap, take, tap } from 'rxjs';
+import { Application } from '../admin/applications/admin-application.model';
+import {
+  RxApplications,
+  RxApplication
+} from '../admin/applications/rx-applications';
+import { Profile } from '../profile/profile.service';
+import { Course, Section } from '../academics/academics.models';
+import {
+  RxCourseList,
+  RxSectionList
+} from '../academics/academics-admin/rx-academics-admin';
+
+@Injectable({ providedIn: 'root' })
+export class ApplicationsService {
+  private applications: RxApplications = new RxApplications();
+  public applications$: Observable<Application[]> = this.applications.value$;
+
+  private user_application: RxApplication = new RxApplication();
+  public user_application$: Observable<Application | null> =
+    this.user_application.value$;
+
+  private new_uta$ = new BehaviorSubject<boolean>(true);
+  public new_uta: Observable<boolean> = this.new_uta$.asObservable();
+
+  private courses: RxCourseList = new RxCourseList();
+  public courses$: Observable<Course[]> = this.courses.value$;
+
+  private sections: RxSectionList = new RxSectionList();
+  public sections$: Observable<Section[]> = this.sections.value$;
+
+  constructor(protected http: HttpClient) {
+    this.initializeApplicationState();
+  }
+
+  /** Returns a list of all Applications
+   * @returns {Observable<Application[]>}
+   */
+  list(): void {
+    this.http
+      .get<Application[]>('/api/applications')
+      .subscribe((applications) => this.applications.set(applications));
+  }
+
+  initializeApplicationState(): void {
+    this.getApplication().subscribe((application) => {
+      this.user_application.set(application);
+      this.new_uta$.next(!application);
+    });
+  }
+
+  getApplication(): Observable<Application | null> {
+    return this.http
+      .get<Application | null>('/api/applications/user')
+      .pipe(
+        tap((application) => console.log('Fetched application:', application))
+      );
+  }
+
+  submitApplication(
+    application: Omit<Application, 'id'>
```

Comment:
Aside: What is `Omit`? We have never used this, seems intriguing

Reply:
Sends the Application object without an id field - I use it here to avoid issues with assigning an id for the frontend model, allowing the backend to set the id instead!